source/solvers/cahn_hilliard.cc

```diff
@@ -1054,6 +1054,22 @@ CahnHilliard<dim>::setup_dofs()
     DoFTools::make_hanging_node_constraints(this->dof_handler,
                                             nonzero_constraints);
 
+    for (auto const &[id, type] :
+         this->simulation_parameters.boundary_conditions.type)
+      {
+        if (type == BoundaryConditions::BoundaryType::periodic)
+          {
+            DoFTools::make_periodicity_constraints(
+              this->dof_handler,
+              id,
+              this->simulation_parameters.boundary_conditions
+                .periodic_neighbor_id.at(id),
+              this->simulation_parameters.boundary_conditions.periodic_direction
+                .at(id),
+              nonzero_constraints);
+          }
+      }
+
     for (auto const &[id, type] :
```

Comment:
Would it not be just simply easier to have to manually specify the periodicity for every physics.
I feel it would be a lot more explicit if we had it this way instead of assuming.